File: c#/prototype/Ejm1/UserAccount.cs
using System;
using System.Collections;
namespace Prototype.Ejm1
{
    /// <summary>
    /// Cuenta de Usuario
    /// </summary>

    public class UserAccount : Prototype
    {
        private string UserName;
        private string Password;
        private string FirstName;
        private string LastName;
        private ArrayList permissions = new ArrayList();

        public UserAccount()
        {

        }
        public override string ToString()
        {
            string res = "UserName:" + UserName + " Password:" + Password + " Nombre:" + FirstName + " Apellidos:" + LastName + " Permisos:";
            foreach (Object obj in permissions)
            {
                res = res + ":" + (string)obj;
            }
            return res;

        }
        /// <summary>
        /// Retorna una copia superficial (shallow copy)
        /// </summary>
        /// <returns>Copia del objeto</returns>
        public override Prototype Clone()
        {
            //El método MemberwiseClone crea una copia superficial
            return (Prototype)this.MemberwiseClone();
        }

        public void SetUserName(string uName)
        {
            UserName = uName;
        }

        public string GetUserName()
        {
            return UserName;
        }

        public void SetPassword(string pwd)
        {
            Password = pwd;
        }

        public string GetPassword()
        {
            return Password;
        }

        public void SetFirstName(string name)
        {
            FirstName = name;
        }

        public string GetFirstName()
        {
            return FirstName;
        }

        public void SetLastName(string name)
        {
            LastName = name;
        }

        public string GetLastName()
        {
            return LastName;
        }

        public void SetPermissions(ArrayList rights)
        {
            permissions = rights;
        }

        public ArrayList GetPermissions()
        {
            return permissions;
        }
    }
}

File: java/prototype/ej1/UserAccount.java
package patrones.prototype.ej1;

import java.util.Vector;
/**
 * Representa una cuenta de Usuario
 *
 */
public class UserAccount implements Cloneable {
	private String userName;
	private String password;
	private String fname;
	private String lname;
	private Vector permissions = new Vector();
	
	public String toString(){
		String res ="UserName:" + userName + " Password:" + password + " Nombre:" + fname + " Apellidos:"+ lname + " Permisos:";
		for (int i=0; i<permissions.size(); i++){
			res = res + ":" + (String)permissions.get(i);
		}
		return res;
			 
	}
	public Object clone() {
		// Shallow Copy
		try {
			return super.clone();
		} catch (CloneNotSupportedException e) {
			return null;
		}
	}

	public void setUserName(String uName) {
		userName = uName;
	}

	public String getUserName() {
		return userName;
	}

	public void setPassword(String pwd) {
		password = pwd;
	}

	public String getPassword() {
		return password;
	}

	public void setFName(String name) {
		fname = name;
	}

	public String getFName() {
		return fname;
	}

	public void setLName(String name) {
		lname = name;
	}

	public String getLName() {
		return lname;
	}

	public void setPermissions(Vector rights) {
		permissions = rights;
	}

	public Vector getPermissions() {
		return permissions;
	}
}
